docs/src/developer/reference/oauth.md

```diff
@@ -0,0 +1,402 @@
+        "d": "..."
+      }
+```
+
+Note that the JWK is a sensitive configuration value, so it is recommended to use Helm's support for managing secrets instead of including it in a plaintext values.yaml file.
+
+### OAuth authorization code, access token, and refresh token expiration
+
+The the OAuth authorization code expiration and access and refresh token expiration can be overridden in the Helm file as follows:
+
+```yaml
+brig:
+  # ...
+  config:
+    # ...
+    optSettings:
+      # ...
+      setOAuthAuthCodeExpirationTimeSecs: 300 # 5 minutes
+      setOAuthAccessTokenExpirationTimeSecs: 300 # 5 minutes
+      setOAuthRefreshTokenExpirationTimeSecs: 14515200 # 24 weeks
+```
+
+### Maximum number of active refresh tokens
+
+The maximum number of active OAuth refresh tokens a user is allowed to have can be configured as follows:
+
+```yaml
+brig:
+  # ...
+  config:
+    # ...
+    optSettings:
+      # ...
+      setOAuthMaxActiveRefreshTokens: 20
+```
+
+## Enable 3rd party apps for teams
+
+3rd party apps are enabled based on the team's payment plan by `ibis`.
+
+### OutlookCalIntegrationConfig
+
+## Implementation details
+
+### Token handling
+
+#### Access token
+
+Payload
+
+{
+  "iss": "https://prod-nginz-https.wire.com",
+  "sub": "7cf24b6c-8c7e-4788-a532-2c998d20ce4a",
+  "aud": "https://prod-nginz-https.wire.com",
+  "exp": 1311281970,
+  "iat": 1311280970,
+  "scope": "write:conversations write:conversations.code"
+}
+
+iss: Wire server
+
+sub: User ID
+
+aud: The resource server
+
+exp: expiration time
+
+iat: issued at time
+
+scope: scopes
+
+Access tokens 
+
+Verification:
+
+- Signature
+- Scope
+- Expiration
+
+#### Refresh token
+
+refresh tokens are assigned to a user and a 3rd party app (OAuth client)
+
+a user can have more than one active refresh token for the same 3rd party app (e.g. they might use multiple devices, replace devices, or run multiple instances of the app somehow)
+
+the maximum number of active refresh tokens per user and app should be limited (what is a sensitive number, 10?)
+
+once a new token is requested and the limit is exceeded, the oldest refresh token will be deleted/invalidated
+
+once a refresh token is used, it will be invalidated and a new refresh token will be generated and returned as part of the response (token rotation)
+
+for now, we will not yet implement re-use detection, but in the future this should be possible
+
+the refresh token is given to the client/app as a signed JWT containing only the refresh token ID
+
+refresh tokens are long-lived, and the expiration should be configurable on the server level (default 3-6 months?)
+
+### Public/private keys
+
+New OAuth JWK keys have to be provided.
+
+Acceptance criteria:
+
+A secure JWK key has been generated
+
+JWK key including the private key has been set in brig secrets for prod and staging
+
+Public JWK key has been set in galley options
+
+OAuth end-to-end test works with new JWK key
+
+### Default configs
```

Comment:
I assume this is still a WIP list of notes?

Reply:
yes